### ISSUE TITLE
ByteMatchers should support variables in "NumBytes" probably in other fields.

### ISSUE THREAD

**Issue by duanehoward** [google/gonids#114]
We currently parse the NumBytes for isdataat, and others as an int.
Some rules seem to use variables here:

```
... byte_extract:4,0,variable,relative,little; content:"foo"; within:3; distance:4; isdataat:variable,relative; ...
```
We currently fail to parse "variable" in isdataat because it's not an int.

1. Validate that this is true for Suricata (documentation doesn't specifically mention this)
2. Migrate to using `string` for the NumBytes field for ByteMatchers.

**Comment by duanehoward** [comment]
Documentation[0] states that variables can be used anywhere an integer is allowed for a modifier. So possibly the most correct thing to do here is to.
  1. Move all 'int' to string.
  2. In all cases validate that the value is either a valid integer *or* a variable that has already been parsed in the rule.

[0] https://suricata.readthedocs.io/en/latest/rules/payload-keywords.html#byte-extract

**Comment by duanehoward** [comment]
Looks like we only need to do this for ByteMatcher types, and `depth`, `offset`, `distance`, `within`
The latter of which are already covered as `ContentOption` uses string for both `Name` and `Value`

From the IRC response:
```
20:29 < jlucovsky0> Variables from byte_extract can be used by: byte_jump, byte_test, byte_test, depth,
                    distance, isdataat, offset and within
20:33 < VictorJ> duane_, ^
```

**Comment by duanehoward** [comment]
Fixed by PR #115
